independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -185,6 +186,30 @@ public List<Resource> generateResources(ReflectionRegistration reflectionRegistr
             ExecutorService executor)
             throws IOException, InterruptedException, ExecutionException {
 
+        List<Map.Entry<String, List<BeanInfo>>> duplicateBeanIds = beanDeployment.getBeans()
```

Comment:
Hm, I think that it would make more sense to move this check in the `arc-processor`, i.e. in the `io.quarkus.arc.processor.BeanDeployment.validateBeans(List<Throwable>, Consumer<BytecodeTransformer>)` method or in the `io.quarkus.arc.processor.BeanProcessor.validate(Consumer<BytecodeTransformer>)`.

Reply:
`BeanDeployment.validateBeans()` sounds like a good spot, will move. Thanks!